applauncher/004-three-by-three.qml

```diff
@@ -0,0 +1,310 @@
+                }
+
+                Rectangle {
+                    id: circle
+
+                    anchors.fill: parent
+                    radius: width / 2
+                    color: !pressAndHoldTimer.running && launcherItem.pressed | fakePressed ?
+                               alb.centerColor(launcherModel.get(currentPressedIndex).filePath) :
+                               "#f4f4f4"
+                    Behavior on color {
+                        PropertyAnimation { target: circle; property: "color"; duration: 70 }
+                    }
+                }
+            }
+
+            DropShadow {
+                anchors.fill: circleWrapper
+                horizontalOffset: 0
+                verticalOffset: 0
+                radius: 8.0
+                samples: 17
+                color: "#80000000"
+                source: circleWrapper
+                cached: true
+            }
+
+            Icon {
+                id: icon
+
+                width: circleWrapper.width * .70
+                height: width
+                anchors.centerIn: circleWrapper
+                color: !pressAndHoldTimer.running && launcherItem.pressed | fakePressed ? "#fff" : "#555"
+                name: model.object.iconId === "" ? "ios-help" : model.object.iconId
+                Behavior on color {
+                    PropertyAnimation { target: icon; property: "color"; duration: 70 }
+                }
+            }
+        }
+
+        Component.onCompleted: {
+            launcherColorOverride = true
+            toLeftAllowed = false
+            toRightAllowed = false
+            toBottomAllowed =  Qt.binding(function() { return !atYBeginning })
+            toTopAllowed = Qt.binding(function() { return !atYEnd })
+            forbidTop = Qt.binding(function() { return !atYBeginning })
+            forbidBottom = false
+            forbidLeft = false
+            forbidRight = false
+        }
+    }
+
+    MouseArea {
+        id: absoluteMouse
+
+        anchors.fill: parent
+        propagateComposedEvents: true
+
+        onPressed: {
+            clickY = Math.round(mouse.y)
+            mouse.accepted = false
+        }
+        onClicked: mouse.accepted = false
+        onReleased: mouse.accepted = false
+        onDoubleClicked: mouse.accepted = false
+        onPositionChanged: mouse.accepted = false
+        onPressAndHold: mouse.accepted = false
+    }
+
+    Rectangle {
+        id: titleShutter
+
+        opacity: 0
+        visible: !root.clickToggle
+        anchors {
+            centerIn: hoverTitle
+            verticalCenterOffset: root.clickY > Dims.h(48) ? -Dims.h(2) : Dims.h(2)
+        }
+        width: root.width
+        height: root.height * .24
+        color: alb.centerColor(launcherModel.get(currentPressedIndex).filePath)
+    }
+
+    Text {
+        id: hoverTitle
+
+        //Offset hoverText and shutter out of view in L290, either top or bottom depending on vertical click position
```

Comment:
I recommend you don't reference a line number. Almost certainly someone will add or remove one line of code someday and the comment will become unreadable.